src/llama.cpp

```diff
@@ -3346,37 +3346,37 @@ static size_t llama_get_device_count(const llama_model & model) {
 static ggml_backend_buffer_type_t llama_default_buffer_type_offload(const llama_model & model, int gpu) {
     ggml_backend_buffer_type_t buft = nullptr;
 
-#if defined(GGML_USE_RPC)
-    int dev_count = (int)llama_get_device_count(model);
     int rpc_count = (int)model.rpc_servers.size();
-    if (gpu >= dev_count - rpc_count) {
-        const char * endpoint = model.rpc_servers[gpu - dev_count + rpc_count].c_str();
+    int local_gpu = gpu - rpc_count;
```

Comment:
I don't think this will work correctly if a list of RPC servers is given in a build without the RPC backend (they should be ignored). The device ids should be from 0 to `llama_get_device_count() - 1`.

Reply:
I think giving a list of RPC servers in non-RPC build should produce an error, i.e. the `--rpc` command line option must not be available.

Reply:
Fixing that in the examples is good, but it still needs to be handled properly for 3rd party applications that use the llama.cpp API directly. `llama_model::rpc_servers` should probably only exist in builds with the RPC backend.

Reply:
I have added `ifdefs` for the `rpc_count`. If we want to remove the existence of `llama_model::rpc_server`, we will need more `ifdefs`.